### PR TITLE
Infrastructure queries (take 2)

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

File: src/routing/approved.rs [hunk 1]
```diff
@@ -993,15 +1015,12 @@ impl Approved {
         }
 
         if join_request.section_key != *self.section.chain().last_key() {
-            let response = BootstrapResponse::Join {
+            let variant = Variant::Rejoin {
```
Comment (maqi) [inline src/routing/approved.rs:1018]: I am still feeling this is bit confusing with the `rejoined node` concept.
I understand here it means `ask the peer to try re-join with new key`.
However, actually the `re-join process` won't help to update the key.
The peer has to go through `re-bootstrap`.

Just wondering, why not use the word of `ReBootstrap`?

File: src/routing/approved.rs [hunk 2]
```diff
@@ -1918,23 +1937,20 @@ impl Approved {
...
-            recipient,
-            message.to_bytes(),
-        ))
+        Ok(Command::send_message_to_node(recipient, message.to_bytes()))
     }
 
     // TODO: consider changing this so it sends only to a subset of the elders
     // (say 1/3 of the ones closest to our name or so)
-    fn send_message_to_our_elders(&self, msg_bytes: Bytes) -> Command {
+    fn send_message_to_our_elders(&self, msg: Bytes) -> Command {
```
Comment (maqi) [inline src/routing/approved.rs:1945]: I see some place change from `msg` to `msg_bytes`, and some place change from `msg_bytes` to `msg`
any guidline to follow?

File: src/routing/command.rs [hunk 3]
```diff
@@ -25,6 +27,7 @@ use std::{
 use tokio::sync::mpsc;
 
 /// Command for node.
+#[allow(clippy::large_enum_variant)]
```
Comment (maqi) [inline src/routing/command.rs:30]: got one removed, here adding a new one.  :)

File: src/routing/approved.rs (hunk 1 above)
Comment (madadam) [reply src/routing/approved.rs:1018]: I see. `Rebootstrap` might work, but it's still not entirely accurate. Let me think about it.

File: src/routing/command.rs (hunk 3 above)
Comment (madadam) [reply src/routing/command.rs:30]: Right. We could either keep the `allow` or `Box` the `message` field in the `SendMessage` variant. I don't have a strong preference either way. What do you think?

File: src/routing/approved.rs (hunk 2 above)
Comment (madadam) [reply src/routing/approved.rs:1945]: I guess `msg` is when it is a concrete type, like say `Message` and `msg_bytes` when it's just `Bytes`. But I'm not sure we are being consistent. Not sure it matters too much though, unless we deal with both `Message` and `Bytes` in the same scope.

File: src/routing/approved.rs (hunk 1 above)
Comment (madadam) [reply src/routing/approved.rs:1018]: How about `JoinRetry` ?

**Review by bochaco — COMMENTED** [review]
This looks good to me @madadam, I'm sending a PR (https://github.com/madadam/sn_routing/pull/4) now which just adds to the client messages test to verify it can receive both on uni and bi streams.

File: src/routing/command.rs (hunk 3 above)
Comment (madadam) [reply src/routing/command.rs:30]: Actually, prefer to keep the `allow`, as boxing the large variant introduces lot of noise all over the codebase.

File: src/routing/approved.rs (hunk 2 above)
Comment (maqi) [reply src/routing/approved.rs:1945]: ok. just being curious as seeing both changes within the same PR  :)

File: src/routing/command.rs (hunk 3 above)
Comment (maqi) [reply src/routing/command.rs:30]: Yeah, I'd prefer leave the `allow` as well.

**Review by maqi — APPROVED** [review]

**Pull request merged**